### PR TITLE
Added quick sort using C

### PR DESCRIPTION
**Pull request by shadowasphodel2919**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added quick sort using C

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

https://github.com/TesseractCoding/NeoAlgo/issues/2238#issue-823716998

### CODE REVIEW

File: C/sort/quickSort.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+of finding a pivot element and dividing the elements in two parts that is, elements
+smaller than pivot are kept on one side and the element larger than the pivot element 
+are placed on the other side of the pivot element.*/
+/*
+Space complexity: O(N log N)
+Time Complexity:
+Best Case: Ω(N log N)		
+Average Case: Θ(N log N)
+Worst Case: O(N^2)
+*/
```
Comment (HarshCasper) [inline C/sort/quickSort.c]: Place this at the end of the code

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C/sort/quickSort.c (hunk 1 above)
Comment (shadowasphodel2919) [reply C/sort/quickSort.c]: Yup I changed it's position.

File: C/sort/quickSort.c [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*Quick Sort is a divide and conquer algorithm which works on the basic principle 
+of finding a pivot element and dividing the elements in two parts that is, elements
+smaller than pivot are kept on one side and the element larger than the pivot element 
+are placed on the other side of the pivot element.*/
```
Comment (ankitaggarwal23) [inline C/sort/quickSort.c]:
```
/*
OUICK SORT:

Quick Sort is a divide and conquer algorithm which works on the basic principle 
of finding a pivot element and dividing the elements in two parts that is, elements
smaller than pivot are kept on one side and the element larger than the pivot element 
are placed on the other side of the pivot element.
*/
```
rewrite it.

File: C/sort/quickSort.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+#include <stdio.h>
+int partitionArray(int[], int, int);
+void quickSort(int[], int, int);
+void printArray(int[], int);
```
Comment (ankitaggarwal23) [inline C/sort/quickSort.c]: remove this.

File: C/sort/quickSort.c [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    for(i = 0; i <size; i++){
+        scanf("%d",&A[i]);
+    }
+
+    printf("\n");
+
+    quickSort(A, 0, size - 1);
+
+    printArray(A, size);
+}
```
Comment (ankitaggarwal23) [inline C/sort/quickSort.c]: shift this main after the function ends

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the changes mentioned

**Comment by shadowasphodel2919** [comment]
I have made the required changes @ankitaggarwal23

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**